### PR TITLE
[indexer] - network_metrics endpoint

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

* added endpoint for TPS and CPS
* refactored checkpoint, moved end of epoch data to epochs table to reduce empty columns
* removed tps from checkpoint

## Test Plan 

Added integration test and tested locally

request
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "method": "suix_getNetworkMetrics",
    "params": []
    }
}
```
response
```
{
    "jsonrpc": "2.0",
    "result": {
        "currentTps": 1.0,
        "tps30Days": 1.4,
        "currentCps": 1.0,
        "cps30Days": 1.4,
        "totalPackages": 3,
        "totalAddresses": 10,
        "totalObjects": 88,
        "currentEpoch": 10,
        "currentCheckpoint": 821
    },
    "id": 1
}
```

### CODE REVIEW

File: crates/sui-open-rpc/spec/openrpc.json [hunk 1]
```diff
@@ -4385,6 +4345,72 @@
           }
         }
       },
+      "NetworkMetrics": {
```
Comment (Jordan-Mysten) [inline crates/sui-open-rpc/spec/openrpc.json:4348]: I'd personally split this into two APIs: one for current data, one for historical (count) data.

**Comment by gegaowp** [comment]
LG overall, had a chat with Patrick and we will make several changes around this PR.

File: crates/sui-open-rpc/spec/openrpc.json (hunk 1 above)
Comment (patrickkuo) [reply crates/sui-open-rpc/spec/openrpc.json:4348]: I am trying not to introduce too many new api just for metrics, also theses metrics are small and the queries are cached, so it's no harm returning all of them in one call.

File: crates/sui-json-rpc-types/src/sui_extended.rs [hunk 2]
```diff
@@ -46,3 +49,47 @@ pub struct EndOfEpochInfo {
     pub total_stake_rewards_distributed: u64,
     pub leftover_storage_fund_inflow: u64,
 }
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct NetworkMetrics {
```
Comment (gegaowp) [inline crates/sui-json-rpc-types/src/sui_extended.rs:55]: lgtm, cc @Jordan-Mysten to confirm.

**Review by gegaowp — APPROVED** [review]
Thanks Patrick for quickly working this out!

**Pull request merged**